src/services/utilities.ts

```diff
@@ -1620,20 +1620,59 @@ namespace ts {
         return position;
     }
 
+    interface RenamedIdentifier {
+        identifier: Identifier;
+    }
+
     /**
      * Creates a deep, memberwise clone of a node with no source map location.
      *
      * WARNING: This is an expensive operation and is only intended to be used in refactorings
      * and code fixes (because those are triggered by explicit user actions).
      */
-    export function getSynthesizedDeepClone<T extends Node | undefined>(node: T, includeTrivia = true): T {
-        const clone = node && getSynthesizedDeepCloneWorker(node as NonNullable<T>);
+    export function getSynthesizedDeepClone<T extends Node | undefined>(node: T, includeTrivia = true, renameMap?: Map<RenamedIdentifier>, nodeIdMap?: Map<boolean>, checker?: TypeChecker, originalType?: Map<Type>): T {
+
+        let clone;
+        if (node && isIdentifier(node!) && renameMap && checker) {
+            const type = checker.getTypeAtLocation(node!);
+            const symbol = checker.getSymbolAtLocation(node!);
+            const renameInfo = symbol && renameMap.get(String(getSymbolId(symbol)));
+
+            if (renameInfo && type && !type.getCallSignatures().length) {  // never rename a function

```

Comment:
I don't understand why this check is here.  Why did the caller put symbol in the map if they didn't want it renamed?